### DISCUSSION TITLE
Retrofit Kit Sensors show as Occupancy Sensors in HomeKit

### DISCUSSION THREAD

**Discussion by GizmoGoody** [tsightler/ring-mqtt#533] [Support Q&A]
I have a mix of Ring contact sensors and Retrofit Kit (zone) sensors. All of them appear as binary sensors in HA. The Ring contact sensors appear properly in HomeKit as "Sensors," but the zones that are connected through the Retrofit Kit appear as "Occupancy" in HomeKit.

Is there a change that could be made to ring-mqtt for HomeKit to be able to see them as "Sensors?"

**Comment by tsightler** [comment]
> Is there a change that could be made to ring-mqtt for HomeKit to be able to see them as "Sensors?"

As they are all just binary sensors what functional difference does this make?  Regardless, you can override the default category directly in Home Assistant on the entity settings page.  Nothing needs to be changed anywhere.

**Reply by GizmoGoody** [reply]
"As they are all just binary sensors what functional difference does this make?"

No functional difference, but the HomeKit UX was confusing at worst, just wrong at best.

"Regardless, you can override the default category directly in Home Assistant on the entity settings page."

Thanks! I did not know this could be changed. All of the Retrofit Kit sensors' "Show as" fields were set to "Safety" by default. I changed them to "Door" or "Window" and reloaded the HomeKit integration. They're all good now.

"Nothing needs to be changed anywhere."

Now I'm just trying to understand, but what assigns the Retrofit Kit zones their default "Show as" category?

**Reply by tsightler** [reply]
Binary sensors in Home Assistant have a [device class](https://www.home-assistant.io/integrations/binary_sensor/#device-class) which mostly just sets the icon and appearance in the HA frontend UI,  It didn't really make sense to use door or window as the default sensor type for a zone, since any given zone could quite literally be a mix of multiple sensors of different classes, so I decided to use the generic "safety" class instead.

I don't know anything about the HomeKit stuff, but I guess it has to map Home Assistant device classes to those available for HomeKit.  Why it would map "safety" to "occupancy" is anyone's guess and really a question for the developers of the HomeKit integration.  However, as it's always been easy to override any entity with whatever class you want, I've never worried myself much about it.

**Reply by GizmoGoody** [reply]
Gotcha. Thanks for the explanation.

FWIW, for the Retrofit Kit, Ring recommends only wiring sensors which don't require power (basically doors and windows) unless the user has a way to provide backup power directly to the sensor. So, smoke alarms, motion sensors, and glass breaks are not what they have in mind.

Anyways, thank you for developing ring-mqtt and your help finding the fix on the entity page.
